src/prefect/agent.py

```diff
@@ -91,6 +91,23 @@ def __init__(
             self.default_infrastructure = Process()
             self.default_infrastructure_document_id = None
 
+    async def _select_default_agent_queues(
+        self, client: PrefectClient, queues: List[WorkQueue]
+    ) -> List[WorkQueue]:
+        """
+        Select only those WorkQueues that match the default work pool type, "prefect-agent"
+        """
+        # get the pools for each matched queue
+        wp_filter = WorkPoolFilter(
+            name=WorkPoolFilterName(any_=[q.work_pool_name for q in queues])
```

Comment:
Is it possible to filter on `type` here as well so we don't need to condition on it later?

e.g.
```suggestion
            name=WorkPoolFilterName(any_=[q.work_pool_name for q in queues]),
            type=WorkPoolFilterType(any_=["prefect-agent"])
```

Reply:
I think I missed the `WorkPoolFilterType` so I definitely want to use that. Thanks for calling this out!